latch/registry/table.py

```diff
@@ -453,6 +473,71 @@ def _add_record_upserts_selection(self, updates: List[l.SelectionNode]) -> None:
 
         updates.append(res)
 
+    # delete record
+
+    def delete_record(self, name: str) -> None:
+        """Delete a record.
+
+        Args:
+            name: Target record name.
+        """
+        self._record_mutations.append(_TableRecordsDeleteData(name))
+
+    def _add_record_deletes_selection(
+        self, deletes: List[_TableRecordsDeleteData], updates: List[l.SelectionNode]
+    ) -> None:
+        if len(deletes) == 0:
+            return
+
+        names: JsonValue = [x.name for x in deletes]
+
+        res = _parse_selection(
+            """
+            catalogMultiDeleteSampleByName(input: {}) {
+                clientMutationId
+            }
+            """
+        )
+        assert isinstance(res, l.FieldNode)
+
+        args = l.ArgumentNode()
+        args.name = _name_node("input")
+        args.value = _json_value(
+            {
+                "argExperimentId": self.table.id,
+                "argNames": names,
+            }
+        )
+
+        res.alias = _name_node(f"del{len(updates)}")
+        res.arguments = tuple([args])
+
+        updates.append(res)
+
+    # transaction
+
+    def _coalesce_mutations(
+        self,
+    ) -> List[Union[List[_TableRecordsUpsertData], List[_TableRecordsDeleteData]]]:
+        output = []
+
+        if len(self._record_mutations) == 0:
+            return output
+
+        cur = [self._record_mutations[0]]
+        for i in range(1, len(self._record_mutations)):
+            next_mut = self._record_mutations[i]
+            if isinstance(next_mut, type(cur[0])):
+                cur.append(next_mut)
+                continue
+
+            output.append(cur)
+            cur = [next_mut]
+
+        output.append(cur)
+
+        return output
```

Comment:
Why is this necessary?

Reply:
Also I don't fully understand what this does -- in my mind this takes a list of [A,A,B,B,A,BBB] and turns it into [[A,A], [B,B], [A], [B,B,B]] -- not sure why this is useful.

Reply:
is useful because each string `[A, A, A, ..., A]` can be made into a single function call, so by coalescing in this way we reduce the size of the graphql query